test.ts

```
@@ -0,0 +1,422 @@
+    SELECT Id, Name, Industry, AnnualRevenue, NumberOfEmployees,
+           BillingStreet, BillingCity, BillingState, BillingPostalCode, BillingCountry,
+           ShippingStreet, ShippingCity, ShippingState, ShippingPostalCode, ShippingCountry,
+           Phone, Website, Description, Type, Rating, AccountSource, OwnerId,
+           CreatedDate, LastModifiedDate, LastActivityDate,
+           (SELECT Id, Name, Email, Phone, Title, LastModifiedDate FROM Contacts),
+           (SELECT Id, Name, StageName, Amount, CloseDate, LastModifiedDate FROM Opportunities),
+           (SELECT Id, CaseNumber, Subject, Status, CreatedDate, LastModifiedDate FROM Cases)
+    FROM Account
+    WHERE Id IN ('${accountIds.join("','")}')
+  `;
+  try {
+    let result = await conn.query<Account>(query);
+    const accounts: Account[] = [];
+    do {
+      accounts.push(...result.records);
+      if (!result.done) {
+        result = await conn.queryMore<Account>(result.nextRecordsUrl);
+      }
+    } while (!result.done);
+    return accounts;
+  } catch (error) {
+    console.error("Error fetching account details from Salesforce:", error);
+    throw error;
+  }
+}
+
+function formatAddress(
+  street?: string,
+  city?: string,
+  state?: string,
+  postalCode?: string,
+  country?: string
+): string {
+  const parts = [street, city, state, postalCode, country].filter(Boolean);
+  return parts.join(", ");
+}
+
+async function upsertDocumentToDustDatasource(account: Account) {
+  const documentId = `account-${account.Id}`;
+  const content = `
+    Account Summary for ${account.Name}
+    Basic Account Details:
+    Company Name: ${account.Name}
+    Industry: ${account.Industry || "N/A"}
+    Annual Revenue: ${
+      account.AnnualRevenue
+        ? `$${account.AnnualRevenue.toLocaleString()}`
+        : "N/A"
+    }
+    Number of Employees: ${account.NumberOfEmployees || "N/A"}
+    Phone: ${account.Phone || "N/A"}
+    Website: ${account.Website || "N/A"}
+    Locations:
```

[EH-003] Empty catch block found. According to our guidelines, errors should be properly handled and not suppressed. Consider logging the error or implementing proper error handling.